Index: Domain/Entities/BaseEntitie.cs
===================================================================
diff --git a/Domain/Entities/BaseEntitie.cs b/Domain/Entities/BaseEntitie.cs
deleted file mode 100644
--- a/Domain/Entities/BaseEntitie.cs	(revision c137a87598605efe43ac22776bc8d353f4c40baa)
+++ /dev/null	(revision c137a87598605efe43ac22776bc8d353f4c40baa)
@@ -1,6 +0,0 @@
-namespace Domain.Entities;
-
-public abstract class BaseEntitie
-{
-    public Guid Id { get; set; }
-}
\ No newline at end of file
Index: Domain/Entities/Category.cs
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>namespace Domain.Entities;\r\n\r\npublic class Category : BaseEntitie\r\n{\r\n    public string Name { get; set; } = string.Empty;\r\n    \r\n    public string? Description { get; set; }\r\n    \r\n    public IEnumerable<Product>? Product { get; set; }\r\n}
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Domain/Entities/Category.cs b/Domain/Entities/Category.cs
--- a/Domain/Entities/Category.cs	(revision c137a87598605efe43ac22776bc8d353f4c40baa)
+++ b/Domain/Entities/Category.cs	(date 1739556495094)
@@ -1,10 +1,10 @@
 namespace Domain.Entities;
 
-public class Category : BaseEntitie
+public class Category : BaseEntity
 {
     public string Name { get; set; } = string.Empty;
     
     public string? Description { get; set; }
     
-    public IEnumerable<Product>? Product { get; set; }
+    public virtual IEnumerable<Product>? Products { get; set; }
 }
\ No newline at end of file
Index: Domain/Entities/Order.cs
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>namespace Domain.Entities;\r\n\r\npublic class Order : BaseEntitie\r\n{\r\n    public Guid OrderItemId { get; set; }\r\n    \r\n    public Guid UserId { get; set; }\r\n    \r\n    public double TotalPrice { get; set; } \r\n    \r\n    public string Status { get; set; } = String.Empty;\r\n    \r\n    public DateTime DeliveryDate { get; set; }\r\n    \r\n    public virtual User? User { get; set; } \r\n    \r\n    public virtual OrderItem? OrderItem { get; set; }\r\n}
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Domain/Entities/Order.cs b/Domain/Entities/Order.cs
--- a/Domain/Entities/Order.cs	(revision c137a87598605efe43ac22776bc8d353f4c40baa)
+++ b/Domain/Entities/Order.cs	(date 1739556495103)
@@ -1,6 +1,6 @@
 namespace Domain.Entities;
 
-public class Order : BaseEntitie
+public class Order : BaseEntity
 {
     public Guid OrderItemId { get; set; }
     
@@ -14,5 +14,5 @@
     
     public virtual User? User { get; set; } 
     
-    public virtual OrderItem? OrderItem { get; set; }
+    public virtual IEnumerable<OrderItem>? OrderItems { get; set; }
 }
\ No newline at end of file
Index: Domain/Entities/OrderItem.cs
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>namespace Domain.Entities;\r\n\r\npublic class OrderItem : BaseEntitie\r\n{\r\n    public Guid ProductId { get; set; }\r\n\r\n    public Guid OrderId { get; set; }\r\n\r\n    public int Amount { get; set; }\r\n    \r\n    public virtual Product? Product { get; set; }\r\n    \r\n    public virtual Order? Order { get; set; }\r\n}\r\n
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Domain/Entities/OrderItem.cs b/Domain/Entities/OrderItem.cs
--- a/Domain/Entities/OrderItem.cs	(revision c137a87598605efe43ac22776bc8d353f4c40baa)
+++ b/Domain/Entities/OrderItem.cs	(date 1739556495115)
@@ -1,6 +1,6 @@
 namespace Domain.Entities;
 
-public class OrderItem : BaseEntitie
+public class OrderItem : BaseEntity
 {
     public Guid ProductId { get; set; }
 
Index: Infrastructure/Infrastructure.csproj
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+><Project Sdk=\"Microsoft.NET.Sdk\">\r\n\r\n    <PropertyGroup>\r\n        <TargetFramework>net9.0</TargetFramework>\r\n        <ImplicitUsings>enable</ImplicitUsings>\r\n        <Nullable>enable</Nullable>\r\n    </PropertyGroup>\r\n\r\n    <ItemGroup>\r\n      <ProjectReference Include=\"..\\Domain\\Domain.csproj\" />\r\n    </ItemGroup>\r\n\r\n</Project>\r\n
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Infrastructure/Infrastructure.csproj b/Infrastructure/Infrastructure.csproj
--- a/Infrastructure/Infrastructure.csproj	(revision c137a87598605efe43ac22776bc8d353f4c40baa)
+++ b/Infrastructure/Infrastructure.csproj	(date 1739556495129)
@@ -10,4 +10,30 @@
       <ProjectReference Include="..\Domain\Domain.csproj" />
     </ItemGroup>
 
+    <ItemGroup>
+      <Compile Remove="Data\Repositories\**" />
+    </ItemGroup>
+
+    <ItemGroup>
+      <EmbeddedResource Remove="Data\Repositories\**" />
+    </ItemGroup>
+
+    <ItemGroup>
+      <None Remove="Data\Repositories\**" />
+    </ItemGroup>
+
+    <ItemGroup>
+      <PackageReference Include="Microsoft.EntityFrameworkCore" Version="9.0.2" />
+      <PackageReference Include="Microsoft.EntityFrameworkCore.Design" Version="9.0.2">
+        <PrivateAssets>all</PrivateAssets>
+        <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
+      </PackageReference>
+      <PackageReference Include="Microsoft.EntityFrameworkCore.Proxies" Version="9.0.2" />
+      <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="9.0.2" />
+      <PackageReference Include="Microsoft.EntityFrameworkCore.Tools" Version="9.0.2">
+        <PrivateAssets>all</PrivateAssets>
+        <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
+      </PackageReference>
+    </ItemGroup>
+
 </Project>
Index: Domain/Entities/Product.cs
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>namespace Domain.Entities;\r\n\r\npublic class Product : BaseEntitie\r\n{\r\n    public string Name { get; set; } = String.Empty;\r\n    \r\n    public string Description { get; set; } = String.Empty;\r\n    \r\n    public double Price { get; set; }\r\n    \r\n    public int Amount { get; set; }\r\n    \r\n    public IEnumerable<Category>? Categories { get; set; }\r\n}
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Domain/Entities/Product.cs b/Domain/Entities/Product.cs
--- a/Domain/Entities/Product.cs	(revision c137a87598605efe43ac22776bc8d353f4c40baa)
+++ b/Domain/Entities/Product.cs	(date 1739556495143)
@@ -1,6 +1,6 @@
 namespace Domain.Entities;
 
-public class Product : BaseEntitie
+public class Product : BaseEntity
 {
     public string Name { get; set; } = String.Empty;
     
@@ -10,5 +10,7 @@
     
     public int Amount { get; set; }
     
-    public IEnumerable<Category>? Categories { get; set; }
+    public virtual IEnumerable<Category>? Categories { get; set; }
+    
+    public virtual IEnumerable<OrderItem>? OrderItems { get; set; }
 }
\ No newline at end of file
Index: Domain/Entities/RefreshToken.cs
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>namespace Domain.Entities;\r\n\r\npublic class RefreshToken\r\n{\r\n    public Guid Token { get; set; }\r\n    \r\n    public DateTime ExpirationDate { get; set; }\r\n}
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Domain/Entities/RefreshToken.cs b/Domain/Entities/RefreshToken.cs
--- a/Domain/Entities/RefreshToken.cs	(revision c137a87598605efe43ac22776bc8d353f4c40baa)
+++ b/Domain/Entities/RefreshToken.cs	(date 1739556495158)
@@ -5,4 +5,6 @@
     public Guid Token { get; set; }
     
     public DateTime ExpirationDate { get; set; }
+    
+    public virtual User? User { get; set; }
 }
\ No newline at end of file
Index: Domain/Entities/User.cs
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>using Domain.Enums;\r\n\r\nnamespace Domain.Entities;\r\n\r\npublic class User : BaseEntitie\r\n{\r\n    public Guid RefreshTokenId { get; set; }\r\n    \r\n    public string PhoneNumber { get; set; } = String.Empty;\r\n    \r\n    public string Password { get; set; } = String.Empty;\r\n    \r\n    public string LastName { get; set; } = String.Empty;\r\n    \r\n    public string FirstName { get; set; } = String.Empty;\r\n    \r\n    public string? MiddleName { get; set; }\r\n    \r\n    public DateTime? BirthDate { get; set; }\r\n    \r\n    public string? Email { get; set; }\r\n    \r\n    public Roles Role { get; set; } \r\n    \r\n    public virtual RefreshToken? RefreshToken { get; set; }\r\n}
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Domain/Entities/User.cs b/Domain/Entities/User.cs
--- a/Domain/Entities/User.cs	(revision c137a87598605efe43ac22776bc8d353f4c40baa)
+++ b/Domain/Entities/User.cs	(date 1739556495175)
@@ -2,7 +2,7 @@
 
 namespace Domain.Entities;
 
-public class User : BaseEntitie
+public class User : BaseEntity
 {
     public Guid RefreshTokenId { get; set; }
     
@@ -23,4 +23,6 @@
     public Roles Role { get; set; } 
     
     public virtual RefreshToken? RefreshToken { get; set; }
+    
+    public virtual IEnumerable<Order>? Orders { get; set; }
 }
\ No newline at end of file
